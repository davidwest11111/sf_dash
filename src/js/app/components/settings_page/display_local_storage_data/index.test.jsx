import React from 'react';
import { shallow } from 'enzyme';
import shallowToJson from 'enzyme-to-json';

import LocalStorage, { BUS_STOP_ROUTE_TAGS_KEY } from 'app/utils/local_storage';

import DisplayLocalStorageData from './index';


describe('outputs the expected tree when', () => {
  let wrapper;

  beforeEach(() => {
    jest.spyOn(LocalStorage, 'get').mockReturnValue(['14|1234']);
  });

  test('when the data is not shown (default)', () => {
    wrapper = shallow((
      <DisplayLocalStorageData localStorageKey={BUS_STOP_ROUTE_TAGS_KEY} />
    ));
  });

  test('when the data is shown', () => {
    wrapper = shallow((
      <DisplayLocalStorageData localStorageKey={BUS_STOP_ROUTE_TAGS_KEY} />
    ));
    wrapper.instance().toggleShow();
  });

  afterEach(() => {
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });
});
