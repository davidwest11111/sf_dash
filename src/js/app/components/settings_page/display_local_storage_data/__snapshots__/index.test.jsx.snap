// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`outputs the expected tree when when the data is not shown (default) 1`] = `
<div
  className=""
>
  <p>
    <Button
      color="primary"
      onClick={[Function]}
      tag="button"
    >
      Show
       LocalStorage Data
    </Button>
  </p>
  <p
    className="small"
  >
    JSON representation of the stored data.
  </p>
</div>
`;

exports[`outputs the expected tree when when the data is shown 1`] = `
<div
  className=""
>
  <p>
    <Button
      color="primary"
      onClick={[Function]}
      tag="button"
    >
      Hide
       LocalStorage Data
    </Button>
  </p>
  <p
    className="small"
  >
    JSON representation of the stored data.
  </p>
  <SyntaxHighlighter
    className="my-3"
    language="json"
    style={
      Object {
        "hljs": Object {
          "background": "#272822",
          "color": "#ddd",
          "display": "block",
          "overflowX": "auto",
          "padding": "0.5em",
        },
        "hljs-addition": Object {
          "color": "#a6e22e",
        },
        "hljs-attribute": Object {
          "color": "#bf79db",
        },
        "hljs-built_in": Object {
          "color": "#a6e22e",
        },
        "hljs-builtin-name": Object {
          "color": "#a6e22e",
        },
        "hljs-bullet": Object {
          "color": "#a6e22e",
        },
        "hljs-class .hljs-title": Object {
          "color": "white",
        },
        "hljs-code": Object {
          "color": "#66d9ef",
        },
        "hljs-comment": Object {
          "color": "#75715e",
        },
        "hljs-deletion": Object {
          "color": "#75715e",
        },
        "hljs-doctag": Object {
          "fontWeight": "bold",
        },
        "hljs-emphasis": Object {
          "color": "#a6e22e",
        },
        "hljs-keyword": Object {
          "color": "#f92672",
          "fontWeight": "bold",
        },
        "hljs-link": Object {
          "color": "#bf79db",
        },
        "hljs-literal": Object {
          "color": "#f92672",
          "fontWeight": "bold",
        },
        "hljs-meta": Object {
          "color": "#75715e",
        },
        "hljs-name": Object {
          "color": "#f92672",
        },
        "hljs-quote": Object {
          "color": "#75715e",
        },
        "hljs-regexp": Object {
          "color": "#bf79db",
        },
        "hljs-section": Object {
          "color": "#a6e22e",
          "fontWeight": "bold",
        },
        "hljs-selector-attr": Object {
          "color": "#a6e22e",
        },
        "hljs-selector-id": Object {
          "fontWeight": "bold",
        },
        "hljs-selector-pseudo": Object {
          "color": "#a6e22e",
        },
        "hljs-selector-tag": Object {
          "color": "#f92672",
          "fontWeight": "bold",
        },
        "hljs-string": Object {
          "color": "#a6e22e",
        },
        "hljs-strong": Object {
          "color": "#f92672",
        },
        "hljs-subst": Object {
          "color": "#a6e22e",
        },
        "hljs-symbol": Object {
          "color": "#bf79db",
        },
        "hljs-tag": Object {
          "color": "#f92672",
        },
        "hljs-template-tag": Object {
          "color": "#a6e22e",
        },
        "hljs-template-variable": Object {
          "color": "#a6e22e",
        },
        "hljs-title": Object {
          "color": "#a6e22e",
          "fontWeight": "bold",
        },
        "hljs-type": Object {
          "color": "#a6e22e",
          "fontWeight": "bold",
        },
        "hljs-variable": Object {
          "color": "#a6e22e",
        },
      }
    }
  >
    ["14|1234"]
  </SyntaxHighlighter>
</div>
`;
